<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
  <meta http-equiv="Content-Style-Type" content="text/css">
  <title></title>
  <meta name="Generator" content="Cocoa HTML Writer">
  <meta name="CocoaVersion" content="1404.47">
  <style type="text/css">
    p.p1 {margin: 0.0px 0.0px 12.0px 0.0px; line-height: 14.0px; font: 12.0px Times; color: #000000; -webkit-text-stroke: #000000}
    p.p3 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 14.0px; font: 12.0px Courier; color: #000000; -webkit-text-stroke: #000000}
    p.p4 {margin: 0.0px 0.0px 0.0px 0.0px; line-height: 14.0px; font: 12.0px Courier; color: #000000; -webkit-text-stroke: #000000; min-height: 14.0px}
    span.s1 {font-kerning: none}
  </style>
</head>
<body>
<p class="p1"><span class="s1">A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many things.<span class="Apple-converted-space"> </span></span></p>
<h2 style="margin: 0.0px 0.0px 14.9px 0.0px; line-height: 22.0px; font: 18.0px Times; color: #000000; -webkit-text-stroke: #000000"><span class="s1"><b>Noncompliant Code Example</b></span></h2>
<p class="p1"><span class="s1">With a maximum number of 4 parameters:</span></p>
<p class="p3"><span class="s1">public function addData(p1, p2 , p3 , p4 , p5 )<span class="Apple-converted-space"> </span></span></p>
<p class="p3"><span class="s1">...</span></p>
<p class="p4"><span class="s1"></span><br></p>
<h2 style="margin: 0.0px 0.0px 14.9px 0.0px; line-height: 22.0px; font: 18.0px Times; color: #000000; -webkit-text-stroke: #000000"><span class="s1"><b>Compliant Solution</b></span></h2>
<p class="p3"><span class="s1">public function addData(p1 , p2 , p3 , p4)<span class="Apple-converted-space"> </span></span></p>
<p class="p3"><span class="s1">...</span></p>
<p class="p4"><span class="s1"></span><br></p>
</body>
</html>
