<p>
A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many things.
</p>

<h2>Noncompliant Code Example</h2>
<p>With a maximum of 3 parameters:</p>
<pre>
 function addData(p1 , p2 , p3 , p4)
 print(p1)
 end
</pre>

<h2>Compliant Solution</h2>
<pre>
 function addData(p1 , p2 , p3)
print()
end
</pre>