<p>Having too many return statements in a function increases the function's essential complexity because the flow of execution is broken each time a return statement is encountered. This makes it harder to read and understand the logic of the function.</p>

<h2>Noncompliant Code Example</h2>

<p>With the default threshold of 3:</p>

<pre>
function myFuncion()        -- NOK
  if (condition1) then
    return true
  else 
    if (condition2) then
      return false
     else 
      return true
   end 
  
  return false
  end


end

</pre>
